### ISSUE TITLE
Incoming channel open timeout while remote peer is signing PSBT from HWW

### ISSUE THREAD

**Issue by asvdf** [ElementsProject/lightning#5192]
### Issue and Steps to Reproduce

1.  have a LND-based node
2. decide to peer with a CLN node
3. decide to go with best practices and use funds from cold wallet, either with `lncli` or `bos` in PSBT mode
4. frantically copy/paste addresses and amounts, scan animated QR's, try and copy back full signed tx
5. notice how you miserably failed as remote side canceled funding before you could complete the operation
6. rinse and repeat with greater conviction but same results (actually remote node will immediately refuse the incoming pending channel for N minutes until it waits again... for about less than 60s)

Would be very useful to implement a sane timeout for manual PSBT signing, a few minutes at least.

**Comment by zerofeerouting** [comment]
It would be great if the timeout was configurable in order to not impact node operators that don't care about the short timeout. Operators like me that would offer the opportunity to fund a channel by hand with a PSBT would be able set a more lenient timeout that gives the peer ample time to fund the transaction.

**Comment by BhaagBoseDK** [comment]
what is the current timeout? Any manual copy paste operation is likely to be allowed a grace of 420s at least to allow for old frail trembling hands to ensure all operations are done without error.

**Comment by asvdf** [comment]
> what is the current timeout? Any manual copy paste operation is likely to be allowed a grace of 420s at least to allow for old frail trembling hands to ensure all operations are done without error.

I could probably even do it in 69s straight if I'm really pumped up, but 210s would be the minimum for healthy operators

**Comment by rustyrussell** [comment]
Um, we don't *have* a timeout there?  Does LND stop answering pings or something, while it's waiting?  That would definitely make us disconnect.  Let me test locally....

**Comment by rustyrussell** [comment]
OK, I hacked our code to put a 200 second sleep in before replying with `funding_created`.  Nada, it just works.

Did you get an error message of some kind?  I suspect LND is failing to respond to pings; we used not to send pings until we had a channel open, but now we send them to all peers.   We can test this, if you want?  Edit line 399 of connectd/multiplex.c:

```
					time_from_sec(15 + pseudorand(30)),
```

Change this to add 60 seconds, so it gives 75-105 seconds instead of 15-45:

```
					time_from_sec(15 + 60 + pseudorand(30)),
```

**Comment by BhaagBoseDK** [comment]
can 60 be configurable in case pasting a PBST takes longer for some with poor hand eye coordination?

**Comment by Roasbeef** [comment]
@asvdf which version of CL are were you using in this issue? IIRC, the latest yet-to-be-released version had some changes to support multiple channels per peer which _might_ be interacting here? 

`lnd` won't auto cancel a reservation as long as it's had a single update (peer responded to our initial message). However this _is_ a timeout with the PSBT flow, since we don't want to lease/lock UTXOs for ever. That's around 10 minutes or so however. 

Re ping messages, we respond to those in a different thread/goroutine, so that should always be pretty quick response wise. 

Logs on either side would likely help us find the culprit here.

**Comment by rustyrussell** [comment]
Good point @Roasbeef .  I was assuming this was an older c-lightning, not bleeding-edge.  Did you already have a channel open with that node (in which case, it's a recent release candidate)?

**Comment by asvdf** [comment]
To make it clearer, I was the channel opener, I use `lnd`, and I tried both `bos open` and `lnd open`, first one will quietly wait until you past signed tx, and error then, while `lnd` will just terminate to cli with a `remote canceled funding` error or something on those notes.
Currently using 14.2 version, while the prospective recipient of my open was zerofeerouting themselves, but I wouldn't know which CL version they use

**Comment by sangbida** [comment]
This issue is being closed as part of issue cleanup.

**Reason**: No recent activity (last updated 1393 da

If you believe this issue should remain open, please comment and we can reopen it.
